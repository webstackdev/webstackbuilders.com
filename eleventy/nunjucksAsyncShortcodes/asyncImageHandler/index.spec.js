/**
 * Integration tests for async image handler
 */
const { asyncImageHandler } = require('./')

describe(``, () => {
  test('Gets correct path for absolute path parameter', async () => {
    // Nunjucks provides shortcodes with a `this` parameter including page url
    const element = await asyncImageHandler.call(
      { page: { url: '/' } },
      '/avatars/test-red-dot.png',
      'test red dot alt text'
    )
    expect(element).toMatchSnapshot()
  })
})
