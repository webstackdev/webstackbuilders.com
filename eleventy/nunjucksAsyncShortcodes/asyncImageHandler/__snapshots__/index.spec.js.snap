// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Gets correct path for absolute path parameter 1`] = `
"
<picture>
  <source
    type=\\"image/webp\\"
    srcset=\\"/images/avatars/test-red-dot-24w.webp 24w, /images/avatars/test-red-dot-400w.webp 400w, /images/avatars/test-red-dot-550w.webp 550w\\"
    sizes=\\"100vw\\"
  >
  <source
    type=\\"image/jpeg\\"
    srcset=\\"/images/avatars/test-red-dot-24w.jpeg 24w, /images/avatars/test-red-dot-400w.jpeg 400w, /images/avatars/test-red-dot-550w.jpeg 550w\\"
    sizes=\\"100vw\\"
  >
  <img
    src=\\"/images/avatars/test-red-dot-550w.jpeg\\"
    width=\\"550\\"
    height=\\"550\\"
    alt=\\"test red dot alt text\\"
    class=\\"\\"
  >
</picture>"
`;
